model-optimizer/extensions/ops/merge.py

```diff
@@ -24,6 +24,7 @@ def merge_infer(node: Node):
         # we infer only through executable input nodes
         inferred_nodes = [n for n in node.in_nodes().values() if n['is_partial_inferred']]
         assert len(inferred_nodes) != 0
+        tensor = inferred_nodes[0]
```

Comment:
Not clear for me how this change helps. Possibly comment in code can help. Looks like shape should be the same for inferred_nodes[0] and inferred_and_exacutable[0]. And I don't understand where I am wrong.

Reply:
I faced a case, when `inferred_nodes[0]` and `inferred_and_executable[0]` are different nodes with different shapes. It happens, because the first of input nodes is a Constant and it is marked as not `executable` ( https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/mo/middle/passes/infer.py#L28 ), only as `is_partial_inferred`.

In addition, in `else` body we are trying to set output value from `inferred_and_executable[0]`, and I guess it will be better to set output shape from this node too.

I am not sure, that I can attach screenshots of the model here to show this case.